data_pipeline/data_store/bq_data_service.py

```diff
@@ -0,0 +1,230 @@
+) -> List[SchemaField]:
+    schema = [SchemaField.from_api_repr(x) for x in json_schema]
+    return schema
+
+
+def get_table_schema_field_names(
+        project_name: str,
+        dataset_name: str,
+        table_name: str
+):
+    client = bigquery.Client()
+
+    dataset_ref = client.dataset(dataset_name, project=project_name)
+    table_ref = dataset_ref.table(table_name)
+    try:
+        table = client.get_table(table_ref)  # API Request
+        return [field.name for field in table.schema]
+    except NotFound:
+        return []
+
+
+def extend_table_schema_with_nested_schema(
+        project_name: str, dataset_name: str,
+        table_name: str, new_fields: list
+):
+    client = bigquery.Client()
+    dataset_ref = client.dataset(dataset_name, project=project_name)
+    table_ref = dataset_ref.table(table_name)
+    table = client.get_table(table_ref)  # Make an API request.
+    original_schema = table.schema
+    original_schema_dict = [
+        schema_field.to_api_repr()
+        for schema_field in original_schema
+    ]
+    new_schema_dict = get_new_merged_schema(
+        original_schema_dict, new_fields
+    )
+
+    new_schema = [
+        SchemaField.from_api_repr(schema_field_dict)
+        for schema_field_dict in new_schema_dict
+    ]
+    table.schema = new_schema
+    client.update_table(table, ["schema"])  # Make an API request.
+
+
+def get_new_merged_schema(
+        existing_schema: list,
+        update_schema: list,
+):
+    new_schema = []
+    existing_schema_dict = {
+        schema_object.get("name").lower(): schema_object
+        for schema_object in existing_schema
+    }
+    update_schema_dict = {
+        schema_object.get("name").lower(): schema_object
+        for schema_object in update_schema
+    }
+    merged_dict = {
+        **update_schema_dict,
+        **existing_schema_dict
+    }
+    set_intersection = (
+        set(existing_schema_dict.keys()).intersection(
+            set(update_schema_dict.keys())
+        )
+    )
+
+    fields_to_recurse = [
+        obj_key
+        for obj_key in set_intersection
+        if existing_schema_dict.get(obj_key).get("fields") and
+        isinstance(existing_schema_dict.get(obj_key).get("fields"), list)
+    ]
+    new_schema.extend(
+        [
+            merged_dict.get(key)
+            for key, value in merged_dict.items()
+            if key not in fields_to_recurse
+        ]
+    )
+    for field_to_recurse in fields_to_recurse:
+        field = existing_schema_dict.get(field_to_recurse).copy()
+        field["fields"] = get_new_merged_schema(
+            existing_schema_dict.get(field_to_recurse).get("fields", []),
+            update_schema_dict.get(field_to_recurse).get("fields", []),
+        )
+        new_schema.append(
+            field
+        )
+
+    return new_schema
+
+
+def generate_schema_from_file(full_temp_file_location):
+    file_reader = open(full_temp_file_location)
+    generator = SchemaGenerator(
```

Comment:
I noticed that `manuscript_id` and the `*person_id` fields have the wrong type (`INTEGER` instead of `STRING`). That is actually also an issue with `manuscript_abstract_keywords`.

See https://github.com/bxparks/bigquery-schema-generator/issues/22

You probably want to set `quoted_values_are_strings`:

https://github.com/bxparks/bigquery-schema-generator/blob/fa00afe1c25bb140683e1a16bfd7586bb45e998b/bigquery_schema_generator/generate_schema.py#L73